### PR TITLE
Make Proxima work with latest Mars

### PR DESCRIPTION
**Pull request by yuyiming**
* The first arg of `TensorSlice` should be a set instead of a list;
* Most chunks in `ProximaBuilder` are bundled to specified workers, but only start-OPs are assigned during analyzing. I assign other chunks;

### CODE REVIEW

File: mars/services/task/analyzer/analyzer.py [hunk 1]
```diff
@@ -303,6 +303,12 @@ def gen_subtask_graph(self) -> SubtaskGraph:
             "Assigned %s start chunks for task %s", len(start_ops), self._task.task_id
         )
 
+        # assign expect workers for successors
+        for chunk in self._chunk_graph.topological_iter():
```
Comment (hekaisheng) [inline mars/services/task/analyzer/analyzer.py:307]: No need to use `topological_iter`, `for chunk in self._chunk_graph` is more efficient。

File: mars/services/task/analyzer/analyzer.py [hunk 2]
```diff
@@ -303,6 +303,12 @@ def gen_subtask_graph(self) -> SubtaskGraph:
             "Assigned %s start chunks for task %s", len(start_ops), self._task.task_id
         )
 
+        # assign expect workers for successors
```
Comment (hekaisheng) [inline mars/services/task/analyzer/analyzer.py:306]: "assign expect workers for those specified with expect_worker" seems clearer.

File: mars/services/task/analyzer/analyzer.py (hunk 1 above)
Comment (chaokunyang) [inline mars/services/task/analyzer/analyzer.py:307]: topological_iter is unnecessary

Comment (yuyiming) [reply mars/services/task/analyzer/analyzer.py]: I've fixed this.

File: mars/services/task/analyzer/analyzer.py (hunk 2 above)
Comment (yuyiming) [reply mars/services/task/analyzer/analyzer.py]: good idea

File: mars/services/task/analyzer/analyzer.py (hunk 1 above)
Comment (yuyiming) [reply mars/services/task/analyzer/analyzer.py]: fixed

**Review by wjsi — APPROVED** [review]
LGTM

**Review by hekaisheng — APPROVED** [review]
LGTM

**Pull request merged**